test/layout/test_units.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import Path
+
+import pytest
+
+from fpdf import FPDF
+from test.conftest import assert_pdf_equal
+
+HERE = Path(__file__).resolve().parent
+
+
+def test_unit_default(tmp_path):
+    """Test creating a PDF with no unit"""
+    pdf = FPDF()
+    pdf.set_font("Helvetica")
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="default")
+    assert_pdf_equal(pdf, HERE / "unit_default.pdf", tmp_path)
+
+
+def test_unit_adjustment(tmp_path):
+    """Test creating a PDF with no unit, then changing to another works"""
+    pdf = FPDF()
+    pdf.set_font("Helvetica")
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="default")
+    pdf.k = 1
+    pdf.add_page()
+    pdf.cell(w=10, h=10, txt="pt")
+    assert_pdf_equal(pdf, HERE / "unit_changed.pdf", tmp_path)
```

Comment:
Seems like the PDf reveals that only the margins are affected by the change on the 2nd page, right?

Reply:
The cell size should have also changed, but it's not noticeable.  However, the margins should not have changed, so I have commented out the test for now.

I have an idea for a fix, but want to do one thing at a time.

Reply:
Test & Fix will be in a separate patch.